org.eclipse.xtext.xbase.lib.tests/src/org/eclipse/xtext/xbase/tests/lib/IterableExtensionsTest.java

```diff
@@ -236,4 +236,29 @@ public Boolean apply(Integer p) {
 		}
 	}
 
+	@Test public void testMap () {
+		ArrayList<String> list = newArrayList("foo", "bar");
+		
+		final Functions.Function1<String, String> function = new Functions.Function1<String, String>() {
+			@Override
+			public String apply(String p) {
+				return "Hello "+p;
+			}
+		};
+		
+		assertEquals(newArrayList("Hello foo", "Hello bar"), newArrayList(IterableExtensions.map(list, function)));
+	}
+
+	@Test public void testFlatMap () {
+		ArrayList<String> list = newArrayList("foo", "bar");
+		
+		final Functions.Function1<String, Iterable<String>> function = new Functions.Function1<String, Iterable<String>>() {
```

Comment:
Please add a test where the function returns a List<R> instead of an Iterable<R>